### ISSUE TITLE
linter.verify() `config` param under- / mis-documented

### ISSUE THREAD

**Issue by jmm** [eslint/eslint#5104]
The [`linter.verify()` docs](https://github.com/eslint/eslint/blob/a9fccf41e5f12610c1c358c86d996c4a28ba3223/docs/developer-guide/nodejs-api.md#linter) provide no explanation of what `config` consists of, and the [source code docs](https://github.com/eslint/eslint/blob/2436cc6c1816a7890e35dab38e609daee84d7530/lib/eslint.js#L591) are incorrect:

> @param {Object} config An object whose keys specify the rules to use.

Please consider:

* Updating the source code docs for 1.x and 2.x to be accurate.

* Updating the end user docs to explain what it consists of. I'm guessing it's a subset of the `CLIEngine` opts. I haven't checked, but I assume the `verify()` docs really mean it when they say "It doesn't do any filesystem operations" (and a later example on the page supports that), meaning that options like `configFile|useEslintrc` wouldn't be applicable.

**Comment by eslintbot** [comment]
@jmm Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Thanks for noticing @jmm

**Comment by BYK** [comment]
working on this.

**Comment by jmm** [comment]
Thanks all!

It's great that you updated the source code docs with the `ESLintConfig` part. [This](https://github.com/eslint/eslint/blob/fd079258631d2f3d7941671654be017858c08fc5/lib/eslint.js#L658) still isn't a great description:

> An object whose keys specify the rules to use.

**Comment by BYK** [comment]
@jmm - Anything you have in mind to make it better/more descriptive?

**Comment by jmm** [comment]
@BYK Is the in-source documentation used to generate anything else? If not, maybe just replace that description with something like "See the ESLintConfig type definition.", since it's right there above it. Or even just "See above"?

**Comment by BYK** [comment]
@jmm - Sure, I'd give that a shot. Thanks for the suggestion!

**Comment by BYK** [comment]
@jmm - Welp, looking at the code again. I think that would be a redundant comment since the type info is also associated with that line. Taking no action for the time being.

**Comment by jmm** [comment]
@BYK Just delete this then (because it's wrong)?:

> An object whose keys specify the rules to use.

**Comment by BYK** [comment]
@jmm - Why do you think it is wrong?

**Comment by BYK** [comment]
Shall we say "A key-value mapping for the rules where keys are rule names and values are the rules' configuration"?

**Comment by jmm** [comment]
@BYK I'm not sure how we have a disconnect on this, since you partially fixed it like this:

```js
// Before 
@param {Object} config

/// after
@param {ESLintConfig}
```

It is an `ESLintConfig` object, not a hash of rule name => rule config. The rules are in `config.rules`. No?

(BTW, what format exactly are you using for the docs in the source [`@param`, etc.]?)

**Comment by BYK** [comment]
@jmm - you're right, sorry. Context switching :) Submitted #5288.

**Comment by BYK** [comment]
>(BTW, what format exactly are you using for the docs in the source [@param, etc.]?)

No sure I understand this question. Can you clarify?

**Comment by jmm** [comment]
@BYK 
> @jmm - you're right, sorry. Context switching :)

No problem. Ah yes, that age old culprit.

>  Submitted #5288.

Thanks!

> No sure I understand this question. Can you clarify?

Sure. I mean is it JSDoc, Doxygen, ...? In particular, is `@param` for sure the right thing [here](https://github.com/eslint/eslint/blob/194b718ad10c24272ee7aaccf3db1a183bc7a19b/lib/eslint.js#L647-L652)? Should it not be `@property` or something?

**Comment by BYK** [comment]
@jmm yup, they should have been `@property` tags, good catch!. Let me update #5288 accordingly.

**Comment by jmm** [comment]
@BYK Thanks!
